includes/cache.php

```diff
@@ -33,6 +33,11 @@ public function __construct($parent) {
         $this->cache_path = wp_upload_dir()['basedir'] . $this->cache_path;
         $this->cache_url  = wp_upload_dir()['baseurl'] . $this->cache_url;
 
+	// Workaround to deal with https-instances that have site_url set to http... (Z#23177854)
+        if (is_ssl()) {
+            $this->cache_url = str_replace('http://', 'https://', $this->cache_url);
+        }
+
```

Comment:
there seems to be a function for doing exactly that - changing the scheme to https:// if is_ssl() is true:

```suggestion
        $this->cache_url  = set_url_scheme(wp_upload_dir()['baseurl'] . $this->cache_url);
```